index.js

```diff
@@ -11,19 +11,42 @@ var chroma = require('chroma-js');
 
 exports.main = function(){
 	var maxTabs = simplePrefs.prefs.maxTabs;
-	var max = (maxTabs > 1) ? maxTabs : 10; // Max at least 2 tabs please
+	var max = (maxTabs > 1) ? maxTabs : 2; // Max at least 2 tabs please
 	var maxPrivateBrowsing = simplePrefs.prefs.maxPrivateBrowsing;
+    var maxOpenInNextWindow = simplePrefs.prefs.maxOpenInNextWindow;
+    var maxExcludePinnedTabs = simplePrefs.prefs.maxExcludePinnedTabs;
 
 	var button = ActionButton({
 		id: 'max-tabs-button',
-	  label: title,
+	    label: title,
 		icon: './icon.svg',
 		disabled: true
 	});
 
+    var calcTabsLen = function(win){
+        let len = win.tabs.length;
+        if(maxExcludePinnedTabs){
+            let pinned = 0;
+            for(let i = 0; i < len; i++){
+                if(win.tabs[i].isPinned)
+                    pinned++;
+            }
+            len -= pinned;
```

Comment:
Perhaps here you can just loop through `win.tabs` and count the unpinned ones instead of getting the `length` first then deduct the pinned ones :sweat_smile: